### PR TITLE
[WIP] Add gallery generator

### PR DESCRIPTION
**Pull request by jrieke**
**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] New Feature
- [ ] Feature Improvment
- [ ] Refactoring
- [ ] Documentation
- [ ] Other, please describe:

**Description:**

Adds a new generator, which generates an image gallery in markdown format. See example on https://github.com/jrieke/best-of-streamlit

**Checklist:**

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: src/best_of/generators/markdown_gallery.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+            project_md = markdown_list.generate_project_md(
+                project, config, labels, generate_body=False
+            )
+            category_md += project_md + "\n"
+        category_md += "</details>\n"
+    # print(category_md)
+    return "<br>\n\n" + category_md
+
+
+def process_md_link(text: str) -> str:
```
Comment (LukasMasuch) [inline src/best_of/generators/markdown_gallery.py:235]: Is it possible to reuse the `process_md_link` of the `markdown_list`? We might also move this to utils.

Comment (jrieke) [reply src/best_of/generators/markdown_gallery.py]: Yup, changed it.

File: README.md [hunk 2]
```diff
@@ -531,6 +531,64 @@ The configuration example above changes the default configuration to show all pr
...
+        <td>Maximum number of projects shown per row.</td>
+        <td><code>3</code></td>
+    </tr>
+    <tr>
+        <td><code>mobile_version</code></td>
+        <td>Whether to create an additional, mobile-optimized version with only one column (i.e. with <code>projects_per_row</code> set to <code>1</code>).</td>
+        <td><code>False</code></td>
+    </tr>
+    <tr>
+        <td><code>mobile_output_path</code></td>
```
Comment (LukasMasuch) [inline README.md:572]: Is this currently used? I think the code only mentions `mobile_output_file`.

Comment (jrieke) [reply README.md]: Upsi, typo, changed it.

**Comment by LukasMasuch** [comment]
@jrieke Looks great! Thanks for contributing :)

**Pull request merged**